.github/workflows/gohai.yml

```diff
@@ -3,28 +3,33 @@ name: "Gohai Test"
 # Only run the tests if pkg/gohai was changed
 on:
   push:
-    branches:
```

Comment:
Similarly to what is done for [serverless integration](https://github.com/DataDog/datadog-agent/blob/main/.github/workflows/serverless-integration.yml#L3-L9), there is probably no need to restrict on branches

Reply:
@chouetz I wrote it that way on a [suggestion](https://github.com/DataDog/datadog-agent/pull/16994#discussion_r1190751633) from @KSerrania to avoid running those tests twice in PRs, the difference with `serverless integration` is that we want the `gohai tests` to run on merges too